src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +435,123 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 

```

Comment:
do you want customer to provider customize parsing for the header string?

Reply:
The customer will be providing one or more custom headers in form of a string that follows this pattern *"{key:value},{key:value}"*
This function validates the input string so that we can be able to split the string and properly add the headers to the WebRequest.Headers property